mat_nn_openmp.hpp

```diff
@@ -33,17 +39,22 @@ double su3_mat_nn(std::vector<site> &a, std::vector<su3_matrix> &b, std::vector<
   d_a = a.data(); len_a = a.size();
   d_b = b.data(); len_b = b.size();
   d_c = c.data(); len_c = c.size();
- 
+
   // Move A and B data to the device, Allocate C data
   double ttotal;
+
+  auto tstart = Clock::now();
+  auto tprofiling = tstart;
+
   #pragma omp target data map(to: d_a[0:len_a], d_b[0:len_b]) map(from: d_c[0:len_c])
```

Comment:
Let's convert this to a standalone target enter data directive to ensure that it happens when it's supposed to.